integration-tests/common/src/test/kotlin/me/tatarka/inject/test/FunctionTest.kt

```diff
@@ -8,72 +10,119 @@ import me.tatarka.inject.annotations.Component
 import me.tatarka.inject.annotations.Inject
 import kotlin.test.Test
 
-@Inject class FunctionFoo
+@Inject
+class FunctionFoo
 
-@Component abstract class FunctionComponent {
+@Component
+abstract class FunctionComponent {
     abstract val fooProvider: () -> FunctionFoo
 }
 
-@Inject class PartialFunctionBar(val foo: FunctionFoo, val name: String)
+@Inject
+class PartialFunctionBar(val foo: FunctionFoo, val name: String)
 
-@Component abstract class PartialFunctionComponent {
+@Component
+abstract class PartialFunctionComponent {
     abstract val barFactory: (name: String) -> PartialFunctionBar
 }
 
-@Inject data class ArityFunctionBar2(val arg1: String, val arg2: String)
+@Inject
+data class ArityFunctionBar2(val arg1: String, val arg2: String)
 
-@Inject data class ArityFunctionBar3(val arg1: String, val arg2: String, val arg3: String)
+@Inject
+data class ArityFunctionBar3(val arg1: String, val arg2: String, val arg3: String)
 
-@Inject data class ArityFunctionBar4(val arg1: String, val arg2: String, val arg3: String, val arg4: String)
+@Inject
+data class ArityFunctionBar4(val arg1: String, val arg2: String, val arg3: String, val arg4: String)
 
-@Component abstract class FunctionArityComponent {
+@Component
+abstract class FunctionArityComponent {
     abstract val arity2: (arg1: String, arg2: String) -> ArityFunctionBar2
     abstract val arity3: (arg1: String, arg2: String, arg3: String) -> ArityFunctionBar3
     abstract val arity4: (arg1: String, arg2: String, arg3: String, arg4: String) -> ArityFunctionBar4
 }
 
-@Inject class NullableFunctionBar(val foo: FunctionFoo?)
+@Inject
+class NullableFunctionBar(val foo: FunctionFoo?)
 
-@Component abstract class NullableFunctionComponent {
+@Component
+abstract class NullableFunctionComponent {
     abstract val barProvider: (FunctionFoo?) -> NullableFunctionBar
 }
 
-@Inject class FunctionBar(val foo: () -> FunctionFoo)
+@Inject
+class FunctionBar(val foo: () -> FunctionFoo)
+
+@Component
+abstract class NestedFunctionComponent {
+    abstract val bar: () -> FunctionBar
+}
+
+
+class MyDep
+
+
+@Inject
+data class Person(val house: House.Factory, val name: String) {
 
-@Component abstract class NestedFunctionComponent {
-   abstract val bar: () -> FunctionBar
+    @Inject
+    class Factory(
+        val create: (String) -> Person
+    )
+}
+
+@Inject
+class House(val bricks: Int) {
+    @Inject
+    class Factory(val create: (Int) -> House)
+}
+
+@Component
+abstract class NestedNestedFunctionComponent {
+    abstract val personFactory: Person.Factory
 }
 
 class FunctionTest {
 
-    @Test fun generates_a_component_that_provides_a_function_that_returns_a_dep() {
+    @Test
+    fun nestedNamesGenerated() {
```

Comment:
minor: could you update the test name to follow the pattern of other tests?

Reply:
Yep, done 👍